src/Http/Http.Abstractions/src/Metadata/IProducesResponseTypeMetadata.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+namespace Microsoft.AspNetCore.Http.Metadata
+{
+    /// <summary>
+    /// Defines a contract for outline the response type returned from an endpoint.
+    /// </summary>
+    public interface IProducesResponseTypeMetadata
+    {
+        /// <summary>
+        /// Gets the optimistic return type of the action.
+        /// </summary>
+        Type? Type { get; }
+
+        /// <summary>
+        /// Gets the HTTP status code of the response.
+        /// </summary>
+        int StatusCode { get; }
+
+        /// <summary>
+        /// Gets the content types supported by the metadata.
+        /// </summary>
+        IReadOnlyCollection<string>? ContentTypes { get; }
```

Comment:
Should this be nullable or empty? Does it mean anything different in this case?

Reply:
I think null more clearly communicates "no content types set by user" which is technically possible for void-returning endpoints where the user does something like `app.MapPost("/process", () => { }).Produces(200)`